rct229/rulesets/ashrae9012019/section5/section5rule39.py

```diff
@@ -0,0 +1,50 @@
+from rct229.rulesets.ashrae9012019.data.schema_enums import schema_enums
+from rct229.rule_engine.partial_rule_definition import PartialRuleDefinition
+from rct229.rule_engine.rule_list_indexed_base import RuleDefinitionListIndexedBase
+from rct229.rule_engine.user_baseline_proposed_vals import UserBaselineProposedVals
+
+SUBSURFACE_DYNAMIC_GLAZING = schema_enums["SubsurfaceDynamicGlazingOptions"]
+
+
+class Section5Rule39(RuleDefinitionListIndexedBase):
+    """Rule 39 of ASHRAE 90.1-2019 Appendix G Section 5 (Envelope)"""
+
+    def __init__(self):
+        super(Section5Rule39, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, False, True),
+            each_rule=Section5Rule39.SubsurfaceRule(),
+            index_rmr="proposed",
+            id="5-39",
+            description="Manually controlled dynamic glazing shall use the average of the minimum and maximum SHGC and VT.",
+            ruleset_section_title="Envelope",
+            standard_section="Section G3.1-5(a)5 Building Envelope Modeling Requirements for the Proposed design",
+            is_primary_rule=False,
+            list_path="ruleset_model_instances[0].buildings[*].building_segments[*].zones[*].surfaces[*].subsurfaces[*]",
+        )
+
+    class SubsurfaceRule(PartialRuleDefinition):
+        def __init__(self):
+            super(Section5Rule39.SubsurfaceRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, False, True),
+                required_fields={"$": ["dynamic_glazing_type"]},
+            )
+
+        def get_calc_vals(self, context, data=None):
+            subsurface_p = context.proposed
+            subsurface_dynamic_glazing_type_p = subsurface_p["dynamic_glazing_type"]
+            return {
+                "subsurface_dynamic_glazing_type_p": subsurface_dynamic_glazing_type_p
+            }
+
+        def get_manual_check_required_msg(self, context, calc_vals=None, data=None):
+            subsurface_p = context.proposed
+            return f'SUBSURFACE {subsurface_p["id"]} INCLUDES MANUALLY CONTROLLED DYNAMIC GLAZING IN THE PROPOSED DESIGN. VERIFY THAT SHGC AND VT WERE MODELED AS THE AVERAGE OF THE MINIMUM AND MAXIMUM SHGC AND VT.'
```

Comment:
1) we usually specify pass/fail/undetermined messages at the top of the code. Please refer to rule 5-38.
2) You may want to use the ```manual_check_required_msg``` attribute in the ```PartialRuleDefinition``` class. rule 5-38 is also a good example.

Reply:
I did it this way to provide the subsurface ID along with the message, rather than a generic message that is the same for every subsurface that has an undetermined outcome. Is there a better way to accomplish this?

Reply:
@JacksonJ-KC the output of the rule check includes the id of the subsurface the evaluation is performed.
Below is an example of the evaluation outcome for a component (in this case, the subsurface).
```
{
                "result": "UNDETERMINED",
                "message": "Some message",
                "id": "subsurface-1"
            }
```
I am fine with your method. But I want to let you know the subsurface ID is inside the output.

Reply:
Oh, excellent! Thanks YJ & WX! I will simplify the message to the more typical method and merge to develop.